api/types/endpoint.go

```diff
@@ -56,14 +56,18 @@ type EndpointCheckRun struct {
 }
 
 type EndpointCheck struct {
-	CheckID string
-	Steps   []EndpointCheckStep
+	CheckID    string
+	Steps      []EndpointCheckStep
+	Status     CheckStatus
+	Conclusion *CheckConclusion `json:"conclusion,omitempty"`
```

Comment:
**thought**: what if this was the same field?

I'm guessing we have something like: 

```
Status:

pending ---> running ---> completed_success
                  |
                   ---> completed_failed
```

If that's the case, we can remove some possible error cases by inclduing these in the same field. We'd never have a Conclusion present, with status not completed.